Fix miniconsole indentation

Harrison-Teeg opened the pull request:
<!-- Keep the title short & concise so anyone non-technical can understand it,
     the title appears in PTB changelogs -->
#### Brief overview of PR changes/additions
Added: setWindowWrapHangingIndent so that main window and miniconsole indentation behaviors can match.
Modified the two TBuffer.append methods to apply indent and hangingIndent.
#### Motivation for adding to Mudlet
Hopefully this resolves https://github.com/Mudlet/Mudlet/issues/5936

/claim #5936

#### Other info (issues closed, discussion etc)
The (modified) motivating test:
```
Geyser.MiniConsole:new({name = "indent-test"})
setWindowWrapIndent("indent-test", 10)
setWindowWrapHangingIndent("indent-test", 5)
setWindowWrap("indent-test", 30)
echo("indent-test", "This is the song that doesn't end, yes it goes on and on my friend, some people started singing it not knowing what it was, and they'll continue singing it forever just because this is the song that doesn't end...")
```
Is now rendered like this:
![image](https://github.com/user-attachments/assets/4a755818-603f-4582-8d9b-bf687cdb82ed)

I see this to be an adequate resolution, but it still does have some minor drawbacks: Multiple echo/print actions to the same line will set the last line in the buffer to be treated as a _firstline_ for the purpose of indentation. This can be an issue if it already has hanging indentation. Some users may find it to be a strange behavior using both indentation and hanging indentation, but I expect this is uncommon.

For example:
```
Geyser.MiniConsole:new({name = "indent-test"})
setWindowWrapIndent("indent-test", 10)
setWindowWrapHangingIndent("indent-test", 5)
setWindowWrap("indent-test", 30)
for i=1,50 do
  cecho("indent-test","<white:red>" .. i)
end
```

ends up looking like this:
![image](https://github.com/user-attachments/assets/03abb4e3-7910-4043-9e85-6a02f2ad228b)

Finally, this is my first PR, apologies if I made any mistakes with the styling/approach. I'm happy for any pointers on how to more effectively contribute.

Commit:
Add indentation to TBuffer::append.

Commit:
Expose wrapHangingIndent for miniconsoles and match indent behavior between miniconsoles and main.

Commit:
Bugfix: Set proper linebreak point for indented lines with no spaces.

Commit:
Remove documentation TODO

Commit:
Fix tiny typo

Harrison-Teeg commented:
I'm not sure what to do about the win32 build. It doesn't seem to have anything to do with my commits but rather something with Mudlet's win32 build environment?

Harrison-Teeg commented:
Stresstest 100,000 before (Mudlet 4.19.1):
![image](https://github.com/user-attachments/assets/43c5bae0-1349-4cfa-b28d-c7e4cc5944d1)
After (Mudlet 4.19.1-testing-pr7697-87bbc7ec):
![image](https://github.com/user-attachments/assets/96509716-2c9c-4764-b2dc-02dca9245c68)

vadi2 committed:
Merge branch 'development' into fix-miniconsole-indent

vadi2 commented:
The 32bit environment is now out of the picture (MSYS2 removed the necessary packages), so I've updated the CI.

Thanks for the PR, will review!

Commit:
Adjust indentation to always use unstyled spaces with the console default background color.

Commit:
Disable miniconsole indentation if the first character of the line is a QChar::Space.

Harrison-Teeg commented:
I've added a commit to resolve the indent+hanging indent issue I mentioned in the original PR, so now this:
```
Geyser.MiniConsole:new({name = "indent-test"})
setWindowWrapIndent("indent-test", 10)
setWindowWrapHangingIndent("indent-test", 5)
setWindowWrap("indent-test", 30)
for i=1,50 do
  cecho("indent-test","<white:red>" .. i)
end
```
Looks like this:
![image](https://github.com/user-attachments/assets/b2290303-4ec4-41cd-858e-507b11421bc9)

Side effect: The buffer append method will never prepend a wrapIndent to lines which start with whitespaces. I think this is an overall better behavior anyway.

The other commit is pretty self explanatory. I felt that it looked bad to use the background color of whatever text was a the end of the line for the indentation spaces, so I've adjusted it to just always use the console's default background color, as can be seen in the screenshot above.


I reran the stresstest using the development branch without my commits for better comparison. Again the updated version ran very slightly faster than the original (44.316s vs 44.499s).

SlySven commented:
> Side effect: The buffer append method will never prepend a wrapIndent to lines which start with whitespaces. I think this is an overall better behavior anyway.

I beg to differ I'd prefer the leading whitespace to be in the text background colour - but what is the consensus on this? @vadi2

Harrison-Teeg commented:
> > Side effect: The buffer append method will never prepend a wrapIndent to lines which start with whitespaces. I think this is an overall better behavior anyway.
> 
> I beg to differ I'd prefer the leading whitespace to be in the text background colour - but what is the consensus on this? @vadi2

I'm happy to implement the background color in whichever way is preferred. I see three options for the indentation bg:
1. Use the default background color of the console, as is now implemented.
2. Use the background color of the first TChar in the line for wrapIndent and wrapHangingIndent like this:
```
    v-- indent bg matching "E"
    Example line wrapped
  to next line
  ^-- hanging indent bg matching "t"
```
3. Use the background color of the first TChar in the line for wrapIndent and the last TChar for hanging:
 ```
    v-- indent bg matching "E"
    Example line wrapped
  to next line         ^-- next hanging indent bg matching "d"
```

Commit:
Bugfix: sneaky missing else was causing crashes

Commit:
Small fix to ensure wrapIndent > wrapAt always disables indentation.

Harrison-Teeg committed:
Merge branch 'development' into fix-miniconsole-indent

Harrison-Teeg commented:
This PR is obsolete, superseded by the wrap overhaul: #7714
